### ISSUE TITLE
a bug that governance data is changed immediately when a node shutdown and starts up when voting is being applied.

### ISSUE THREAD

**Issue by yoomee1313** [klaytn/klaytn#1106]
**Describe the bug**
Governance data is altered immediately when a node shutdown and starts up when voting is being applied. This is not an intended behavior.
`Voting is being applied` means a vote has been applied to a governance db, but an epoch has not passed. So, governance data should not be changed. 

**How to reproduce**
Step1. deploy a test network.
Step2. call vote api which tries to alter a governance value.
Step3. shutdown a node and starts up.
Step4. check governance data immediately.

**Expected behavior**
Governance data is not altered immediately.

**Environment (please complete the following information)**
 - Klaytn version or git revision that exhibits the issue `$ ken version`
 - go version `$ go version`
 - OS and its version [e.g. macOS/10.14]

**Comment by aeharvlee** [comment]
[Issue Report]
A panic occurred while testing with the CN1 - PN1 - EN1 configuration built with the latest upstream/dev branch locally.
The problem occurred in the `gov.actualGovernanceBlock.Load().(uint64)` part within the [Governance.UpdateCurrentSet](https://github.com/klaytn/klaytn/blob/d85dd8871094029468608881b112edacec22f1c9/governance/default.go#L810-L818) method.
`actualGovernanceBlock` is nil, but immediately after loading it, a panic occurred while trying to cast it to uint64 .
It seems to set the value to actualGovernanceBlock within the [Governance.initializeCache](https://github.com/klaytn/klaytn/blob/d85dd8871094029468608881b112edacec22f1c9/governance/default.go#L614-L650) function.
I thought we needed to check this part, so I shared it.

The reproduction process is as follows. (It is not fully reproduced.. It is intermittent. I can't figure out the root cause either :cry: )
* Start the network by running `3_ccstart.sh`
* After the epoch block cycle has elapsed, the network is shut down by executing `4_ccstop.sh`
* After repeating the above process about 3 times, an error occurs.

The genesis.json I used to configure the node I used is as follows.
```json
{
    "config": {
        "chainId": 191919,
        "istanbulCompatibleBlock": 0,
        "londonCompatibleBlock": 0,
        "istanbul": {
            "epoch": 30,
            "policy": 2,
            "sub": 1
        },
        "unitPrice": 25000000000,
        "deriveShaImpl": 2,
        "governance": {
            "governingNode": "0x9712f943b296758aaae79944ec975884188d3a96",
            "governanceMode": "single",
            "reward": {
                "mintingAmount": 9600000000000000000,
                "ratio": "34/54/12",
                "useGiniCoeff": false,
                "deferredTxFee": true,
                "stakingUpdateInterval": 60,
                "proposerUpdateInterval": 30,
                "minimumStake": 5000000
            }
        }
    },
    "timestamp": "0x6020908f",
    "extraData": "0x0000000000000000000000000000000000000000000000000000000000000000f85ad5949712f943b296758aaae79944ec975884188d3a96b8410000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0",
    "governanceData": null,
    "blockScore": "0x1",
    "alloc": {
        "0000000000000000000000000000000000000400": {
            "code": "{생략}...",
            "balance": "0x0"
        },
        "9712f943b296758aaae79944ec975884188d3a96": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        },
        "ca7a99380131e6c76cfa622396347107aeedca2d": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        },
        "3e2ac308cd78ac2fe162f9522deb2b56d9da9499": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        },
        "f7ccb6e6ef8386582c917604c0983d70dd555c0c": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        },
        "e19a5e72e8448978dbbaee284a140f67a08ac2a4": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        },
        "2eaad2bf70a070aaa2e007beee99c6148f47718e": {
            "balance": "0x446c3b15f9926687d2c40534fdb564000000000000"
        }
    },
    "number": "0x0",
    "gasUsed": "0x0",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
}
```

And below is the error log.
```
CN log >

INFO[02/14,00:22:51 +09] [40] Starting P2P networking
INFO[02/14,00:22:51 +09] [40] Started P2P server                        id=9419498985bfb636 multichannel=false
INFO[02/14,00:22:51 +09] [40] RLPx listener up                          self="kni://9419498985bfb636a90b97b352bffac514771baa2a30a621a1569ce067ab9227c2f7f383d97e714469c4995a7e89fb15c53b7d65b87c9548b37cc3e434b1c79e@[::]:30303?discport=0"
INFO[02/14,00:22:51 +09] [44] Start listening chain head event to update stakingInfoCache.
INFO[02/14,00:22:51 +09] [40] IPC endpoint opened                       url=/Users/denver.lee/klaytn-network-local/cn/data/klay.ipc
INFO[02/14,00:22:51 +09] [50] Starting mining operation
WARN[02/14,00:22:51 +09] [18] Use explicit addresses! Referring to accounts by order in the keystore folder is dangerous and will be deprecated!
panic: interface conversion: interface {} is nil, not uint64

goroutine 277 [running]:
github.com/klaytn/klaytn/governance.(*Governance).UpdateCurrentSet(0xc00012d560, 0x41a)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/governance/default.go:813 +0x145
github.com/klaytn/klaytn/consensus/istanbul/backend.(*Snapshot).apply(0xc016ca2100, 0xc00018c000, 0xf3, 0x100, 0xc00012d560, 0x8a7596b243f91297, 0x845897ec4499e7aa, 0x963a8d18, 0x2, 0x5bba188, ...)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/snapshot.go:182 +0x9a6
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).snapshot(0xc0000f6000, 0x5bba188, 0xc000bc7800, 0x400, 0xc1dd5ef54684df4e, 0x2acbfe9c21edbe22, 0x6ef901ba3dee6ed9, 0x507b89d907c237fc, 0x0, 0x0, ...)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/engine.go:705 +0x3d8
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).getValidators(0xc0000f6000, 0x4f3, 0x4961058eb564d170, 0xfa53827174708a2c, 0x77d77e72022f7a50, 0x41d85dbe7f75728e, 0x0, 0xc016c83be8)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/backend.go:396 +0x8a
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).Validators(0xc0000f6000, 0x5bbd2f0, 0xc000161490, 0xc000444f00, 0x0)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/backend.go:152 +0xb0
github.com/klaytn/klaytn/consensus/istanbul/core.(*core).startNewRound(0xc0000f62c0, 0xc00009c260)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/core/core.go:257 +0x11f4
github.com/klaytn/klaytn/consensus/istanbul/core.(*core).Start(0xc0000f62c0, 0x1, 0xc0007d0420)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/core/handler.go:31 +0x37
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).Start(0xc0000f6000, 0x5bba188, 0xc000bc7800, 0xc0008b6228, 0xc0008b6240, 0x0, 0x0)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/engine.go:587 +0x127
github.com/klaytn/klaytn/work.(*worker).start(0xc0000f6840)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/work/worker.go:234 +0x2b3
github.com/klaytn/klaytn/work.(*Miner).Start(0xc00021e5f0)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/work/work.go:157 +0x5c
created by github.com/klaytn/klaytn/node/cn.(*CN).StartMining
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/node/cn/backend.go:603 +0x55



EN log >

INFO[02/14,00:22:52 +09] [45] Starting P2P networking
INFO[02/14,00:22:52 +09] [45] Started P2P server                        id=ae50a40728759413 multichannel=false
INFO[02/14,00:22:52 +09] [45] RLPx listener up                          self="kni://ae50a407287594135df1f57eb6ddb95495ee246866239c231a520cc7e1fabd4ba6e270bf7d152bc8327f4aa4010fd2b0a174db7bf994815ce09e7d9cc00e40a5@[::]:50505?discport=0"
INFO[02/14,00:22:52 +09] [40] IPC endpoint opened                       url=/Users/denver.lee/klaytn-network-local/en/data/klay.ipc
INFO[02/14,00:22:52 +09] [40] FastHTTP endpoint opened                  url=http://0.0.0.0:8551                                     cors=* vhosts=*
INFO[02/14,00:22:52 +09] [40] FastWebSocket endpoint opened             url=ws://0.0.0.0:8552
WARN[02/14,00:22:52 +09] [18] Use explicit addresses! Referring to accounts by order in the keystore folder is dangerous and will be deprecated!
panic: interface conversion: interface {} is nil, not uint64

goroutine 342 [running]:
github.com/klaytn/klaytn/governance.(*Governance).UpdateCurrentSet(0xc00014bc20, 0x41a)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/governance/default.go:813 +0x145
github.com/klaytn/klaytn/consensus/istanbul/backend.(*Snapshot).apply(0xc00027c080, 0xc016bbd000, 0xf3, 0x100, 0xc00014bc20, 0xb0d55f74f56f779d, 0xf38f0059325f45df, 0x95ba0ecd, 0x2, 0x5bba188, ...)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/snapshot.go:182 +0x9a6
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).snapshot(0xc0001682c0, 0x5bba188, 0xc0004f2c00, 0x400, 0xc1dd5ef54684df4e, 0x2acbfe9c21edbe22, 0x6ef901ba3dee6ed9, 0x507b89d907c237fc, 0x0, 0x0, ...)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/engine.go:705 +0x3d8
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).getValidators(0xc0001682c0, 0x4f3, 0x4961058eb564d170, 0xfa53827174708a2c, 0x77d77e72022f7a50, 0x41d85dbe7f75728e, 0x0, 0xc000757be8)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/backend.go:396 +0x8a
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).Validators(0xc0001682c0, 0x5bbd2f0, 0xc000a81810, 0xc016cb0180, 0x0)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/backend.go:152 +0xb0
github.com/klaytn/klaytn/consensus/istanbul/core.(*core).startNewRound(0xc000168420, 0xc000146240)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/core/core.go:257 +0x11f4
github.com/klaytn/klaytn/consensus/istanbul/core.(*core).Start(0xc000168420, 0x1, 0xc000a8d080)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/core/handler.go:31 +0x37
github.com/klaytn/klaytn/consensus/istanbul/backend.(*backend).Start(0xc0001682c0, 0x5bba188, 0xc0004f2c00, 0xc00000e360, 0xc00000e3a8, 0x0, 0x0)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/consensus/istanbul/backend/engine.go:587 +0x127
github.com/klaytn/klaytn/work.(*worker).start(0xc000168840)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/work/worker.go:234 +0x2b3
github.com/klaytn/klaytn/work.(*Miner).Start(0xc0000c8e60)
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/work/work.go:157 +0x5c
created by github.com/klaytn/klaytn/node/cn.(*CN).StartMining
	/Users/denver.lee/krustuniverse/dopr/dynamic-and-access/klaytn/node/cn/backend.go:603 +0x55
```

cc. @yoomee1313

**Comment by yoomee1313** [comment]
@aeharvlee thanks for reporting that problem. Unfortunately, I couldn't reproduce it, but i can check there's problem with initializing `g.actualGovernanceBlock`. It is caused by recent PR that i merged, so i made a fix PR (#1168).

**Comment by yoomee1313** [comment]
as https://github.com/klaytn/klaytn/pull/1168 is merged, the issue is closed.
